### PR TITLE
Update SDK nugets and a Moq-ify a test

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs [hunk 1]
```diff
@@ -89,30 +89,33 @@ public async Task RetryDelegatingHandler_DoesNotRetryOnNotSupportedException()
         {
             // arrange
             int callCounter = 0;
-
-            PipelineContext contextMock = Substitute.For<PipelineContext>();
-            contextMock.ConnectionStatusChangeHandler = (connectionStatusInfo) => { };
-            IDelegatingHandler nextHandlerMock = Substitute.For<IDelegatingHandler>();
             var message = new TelemetryMessage(new byte[] { 1, 2, 3 });
-            nextHandlerMock.OpenAsync(Arg.Any<CancellationToken>()).Returns(Task.CompletedTask);
+
```
Comment (drwill-ms) [inline iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs:93]: I find more and more tests using NSubstitute fail occasionally. When one fails, I update it to use Moq.

Soon, we should make a pass and proactively update the remaining ones and remove NSubstitute as a dependency.

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs [hunk 2]
```diff
@@ -89,30 +89,33 @@ public async Task RetryDelegatingHandler_DoesNotRetryOnNotSupportedException()
...
+            PipelineContext contextMock = new PipelineContext();
```
Comment (patilsnr) [inline iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs:94]: `var contextMock = new PipelineContext();`?

**Review by patilsnr — APPROVED** [review]
Optimistic sign off

**Comment by drwill-ms** [comment]
Gates passed and one small change pushed.

**Pull request merged**